services/venice-server/src/main/java/com/linkedin/venice/stats/AggServerHttpRequestStats.java

```diff
@@ -36,9 +39,20 @@ public AggServerHttpRequestStats(
         unregisterMetricForDeletedStoreEnabled);
   }
 
+  public void setStoreStat(String storeName) {
+    this.serverHttpRequestStats = super.getStoreStats(storeName);
+  }
+
+  ServerHttpRequestStats getStoreStat(String storeName) {
+    if (serverHttpRequestStats == null || !storeName.equals(serverHttpRequestStats.getStoreName())) {
+      this.serverHttpRequestStats = super.getStoreStats(storeName);
+    }
+    return serverHttpRequestStats;
+  }
```

Comment:
Hmm... is this safe? I expect that many Netty threads would concurrently interact with this object, and mutating it like this does not seem right in a multi-threaded context. I think we could leave this part unchsnged and take it up in a separate PR. LMK what you think.